libbeat/monitoring/report/elasticsearch/client.go

```diff
@@ -245,25 +251,25 @@ func getMonitoringIndexName() string {
 	return fmt.Sprintf(".monitoring-beats-%v-%s", version, date)
 }
 
-func logBulkFailures(result esout.BulkResult, events []report.Event) {
-	reader := esout.NewJSONReader(result)
-	err := esout.BulkReadToItems(reader)
+func logBulkFailures(result esclientleg.BulkResult, events []report.Event, logger *logp.Logger) {
```

Comment:
oh, I wasn't aware the the monitoring reporter also relies on the 'low-level' bulk result. Maybe we can have the monitoring reported use the other bulk publishing methods and parse the JSON directly. Then we won't need to keep and export BulkResult. Monitoring normally has one event only. We only use bulk, because the x-pack endpoint requires us to do so.

Reply:
Okay, I will look into this. IIRC the monitoring reporter was parsing the bulk results slightly differently than the ES output but lets see if we can achieve more reusability here.

Reply:
Fixed in d76a4bdd7.